### DISCUSSION TITLE
write text to video

### DISCUSSION THREAD

**Discussion by sugizo** [python-ffmpegio/python-ffmpegio#40] [Q&A]
**objective**
write text to video, e.g. using cv2.putText

**code**
```
import ffmpegio
import cv2
input = 'http://195.196.36.242/mjpg/video.mjpg'
output = 'rtsp.mp4'
i = 0
text = 'stifix'
font = cv2.FONT_HERSHEY_COMPLEX
org = (00, 185)
fontScale = 1
color = (0, 0, 128)
thickness = 2
with ffmpegio.open(input, 'rv', s = (720, -1) ) as fin, ffmpegio.open(output, 'wv', rate_in = fin.rate, overwrite = True) as fout:
    for frames in fin:
        put_text = cv2.putText(frames, text, org, font, fontScale, color, thickness, cv2.LINE_AA, False)
        fout.write(put_text)
        if i == 9: # 100 = 4 secs
            break
        i += 1
```

**result**
no error occured 
but output video have no text written

**expected result** 
video output have text written

best regards

**Comment by tikuma-lsuhsc** [comment]
Since no error, I assume that `put_text` is a `uint8` dtype. Correct?

Did you check what `put_text` frame looks like?

You can use Matplotlib to display the frame.

If you never used it, `pip install matplotlib PyQt5` to install it then

```python
from matplotlib import pyplot as plt

...

plt.imshow(put_text)
plt.show()
```

If the displayed frame has no text, then it's not ffmpegio issue. Consult openCV guys as I'm not familiar with it.

As customary, I must let you know that you can also add text with FFmpeg filter: [`drawtext`](https://ffmpeg.org/ffmpeg-filters.html#drawtext-1)

**Comment by sugizo** [comment]
**objective**
change below code into Stream I/O

**code drawtext work**
```
from ffmpegio import ffmpeg
input = 'input_ss_t.mp4'
output = 'drawtext.mp4'
ffmpeg("-y -i %s -vf \"[v0]drawtext = text = 'text_center' : fontcolor = red : fontfile = comic_sans.ttf : fontsize = 32 : x = (w - text_w) / 2 : y = (h - text_h) / 2 : enable = 'between(t, 0, 3)'[v1] \" %s" % (input, output) )
```

**question**
how to change above code into Stream I/O style ?

thanks n best regards
